### PR TITLE
[libjpeg-turbo] Use single-thread compilation when compiling static libraries under windows

### LINKED ISSUE #33220: [libjpeg-turbo] build failure
### Operating system

Windows 11

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
vcpkg install libjpeg-turbo:x64-Windows
```

### Failure logs

Computing installation plan...
The following packages will be built and installed:
    libjpeg-turbo:x64-windows -> 2.1.5.1#2
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\吴\AppData\Local\vcpkg\archives in 110 us. Use --debug to see more details.
Installing 1/1 libjpeg-turbo:x64-windows...
Building libjpeg-turbo:x64-windows...
-- Using cached libjpeg-turbo-libjpeg-turbo-2.1.5.1.tar.gz.
error: building libjpeg-turbo:x64-windows failed with: BUILD_FAILED
Elapsed time to handle libjpeg-turbo:x64-windows: 370 ms

### Additional context

both x64 and x86 version are unavailable

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/33220
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
x64-windows-static
```

### CODE REVIEW

**Comment by Neumann-A** [comment]
This solution doesn't make sense for me given the observed error.

File: ports/libjpeg-turbo/portfile.cmake [hunk 1]
```diff
@@ -36,9 +36,13 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         jpeg7 WITH_JPEG7
         jpeg8 WITH_JPEG8
 )
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    set(COMPILE_THREAD DISABLE_PARALLEL_CONFIGURE)
```
Comment (Neumann-A) [inline ports/libjpeg-turbo/portfile.cmake:40]: This is not single-thread compilation. This is just disabling parallel configure which is only an issue if the SOURCE_DIR is written to by both configure commands. This however was not observed

Comment (jimwang118) [reply ports/libjpeg-turbo/portfile.cmake:40]: According to this [reply](https://github.com/microsoft/vcpkg/issues/33220#:~:text=Use%20cmd%20rather%20than%20powershell%3A%0A1.set%20VCPKG_MAX_CONCURRENCY%3D1%0A2.vcpkg%20install%20...), this problem is caused by parallel construction while reading and writing source.

Comment (Neumann-A) [reply ports/libjpeg-turbo/portfile.cmake:40]: > According to this https://github.com/microsoft/vcpkg/issues/33220#:~:text=Use%20cmd%20rather%20than%20powershell%3A%0A1.set%20VCPKG_MAX_CONCURRENCY%3D1%0A2.vcpkg%20install%20..., this problem is caused by parallel construction while reading and writing source.

No that just tells you that `-j 1` was used. That is not the same as `DISABLE_PARALLEL_CONFIGURE`

File: ports/libjpeg-turbo/portfile.cmake [hunk 2]
```diff
@@ -53,7 +53,12 @@ vcpkg_cmake_configure(
         WITH_CRT_DLL
 )
 
-vcpkg_cmake_install()
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    vcpkg_cmake_install(DISABLE_PARALLEL)
+else()
```
Comment (Neumann-A) [inline ports/libjpeg-turbo/portfile.cmake:58]: I don't believe this is the solution. Its just fixing one of the symptoms. vcpkg CI would have seen this problem already if it was a real issue. The user probably just has AV messing around with the compiled files.

Comment (jimwang118) [reply ports/libjpeg-turbo/portfile.cmake:58]: So you're saying it's just an occasional problem due to antivirus software.

**Comment by BillyONeal** [comment]
I agree, this doesn't seem to be related.

**Pull request closed without merging**